examples/mxnet_mnist.py

```diff
@@ -0,0 +1,127 @@
+# Step 1: import required packages
+import os
+import argparse
+import logging
+import mxnet as mx
+import horovod.mxnet as hvd
+from mxnet.test_utils import get_mnist_ubyte
+
+# Step 1: Initialize horovod
+hvd.init()
+
+# Training settings
+parser = argparse.ArgumentParser(description='MXNet MNIST Example')
+parser.add_argument('--batch-size', type=int, default=64,
+                    help='training batch size (default: 64)')
+parser.add_argument('--dtype', type=str, default='float32',
+                    help='training data type (default: float32)')
+parser.add_argument('--gpus', type=str, default='0',
+                    help='number of gpus to use (default: 0)')
+parser.add_argument('--epochs', type=int, default=10,
+                    help='number of training epochs (default: 10)')
+parser.add_argument('--lr', type=float, default=0.05,
+                    help='learning rate (default: 0.05)')
+parser.add_argument('--momentum', type=float, default=0.5,
+                    help='SGD momentum (default: 0.5)')
+args = parser.parse_args()
+
+logging.basicConfig(level=logging.INFO)
+logging.info(args)
+
+# Horovod: pin GPU to local rank.
+context = mx.cpu() if args.gpus is None or args.gpus == '0' \
+                   else mx.gpu(hvd.local_rank())
+
+# Step 2: data loading
+
+# Download mnist data set using get_mnist_ubyte()
```

Comment:
Is it possible to have a directory per rank that would eliminate data contention?

Or maybe if the directory is not found you could print a message with a full `python -c "..."` command to run to download MNIST files?